editor/property_editor.cpp

```diff
@@ -3642,8 +3646,12 @@ void PropertyEditor::update_tree() {
 						item->set_tooltip(1, res->get_name() + " (" + res->get_class() + ")");
 					}
 					if (!res->is_class("Texture")) {
-						//texture already previews via itself
-						EditorResourcePreview::get_singleton()->queue_edited_resource_preview(res, this, "_resource_preview_done", item->get_instance_ID());
+						if (res->is_class("Script")) {
+							item->set_text(1, res->get_path().get_file());
+						} else {
+							//texture already previews via itself
```

Comment:
Leave this comment where it was, there is no point in moving it around.
Also, **might** look better as: (but I'm not sure if it is much of an improvement...)
```cpp
if (res->is_class("Texture")) {
  //texture already previews via itself
} else if (res->is_class("Script")) {
  item->set_text(1, res->get_path().get_file());
} else {
  EditorResourcePreview::get_singleton()->queue_edited_resource_preview(res, this, "_resource_preview_done", item->get_instance_ID());
}
```
Or:
```cpp
if (res->is_class("Script")) {
  item->set_text(1, res->get_path().get_file());
} else if (!res->is_class("Texture")) {
  //texture already previews via itself

  EditorResourcePreview::get_singleton()->queue_edited_resource_preview(res, this, "_resource_preview_done", item->get_instance_ID());
}
```